youtube_dl/jsinterp.py

```diff
@@ -877,12 +884,35 @@ def eval_method():
                     self.interpret_expression(v, local_vars, allow_recursion)
                     for v in self._separate(arg_str)]
 
-                if obj == compat_str:
+                # Fixup prototype call
+                if isinstance(obj, type):
+                    new_member, rest = member.partition('.')[0::2]
```

Comment:
why not 
```suggestion
                    new_member, _, rest = member.partition('.')
```

Reply:
Call me a slicer rather than a destructurer. Also that's what I had in my somewhat similar code before I applied yt-dlp/yt-dlp#10392.